sql/parse/parse.go

```diff
@@ -1354,6 +1358,75 @@ func selectExprsToExpressions(ctx *sql.Context, se sqlparser.SelectExprs) ([]sql
+func StringToColumnDefaultValue(ctx *sql.Context, exprStr string) (*sql.ColumnDefaultValue, error) {
+	// all valid default expressions will parse correctly with SELECT prepended, as the parser will not parse raw expressions
+	stmt, err := sqlparser.Parse("SELECT " + exprStr)
+	if err != nil {
+		return nil, err
+	}
+	//TODO: change these from fmt.Errorf
+	parserSelect, ok := stmt.(*sqlparser.Select)
+	if !ok {
+		return nil, fmt.Errorf("DefaultStringToExpression expected sqlparser.Select but received %T", stmt)
+	}
+	if len(parserSelect.SelectExprs) != 1 {
+		return nil, fmt.Errorf("default string does not have only one expression")
+	}
+	aliasedExpr, ok := parserSelect.SelectExprs[0].(*sqlparser.AliasedExpr)
+	if !ok {
+		return nil, fmt.Errorf("DefaultStringToExpression expected *sqlparser.AliasedExpr but received %T", parserSelect.SelectExprs[0])
+	}
+	parsedExpr, err := exprToExpression(ctx, aliasedExpr.Expr)
+	if err != nil {
+		return nil, err
+	}
+	// The literal and expression distinction seems to be decided by the presence of parentheses, even for defaults like NOW() vs (NOW())
+	// 2+2 would evaluate to a literal under the parentheses check, but will have children due to being an Arithmetic expression, thus we check for children.
+	return ExpressionToColumnDefaultValue(ctx, parsedExpr, len(parsedExpr.Children()) == 0 && !strings.HasPrefix(exprStr, "("))
+}
+
+// ExpressionToColumnDefaultValue takes in an Expression and returns the equivalent ColumnDefaultValue if the expression
+// is valid for a default value. If the expression represents a literal (and not an expression that returns a literal, so "5"
+// rather than "(5)"), then the parameter "isLiteral" should be true.
+func ExpressionToColumnDefaultValue(ctx *sql.Context, inputExpr sql.Expression, isLiteral bool) (*sql.ColumnDefaultValue, error) {
+	expr, err := expression.TransformUp(inputExpr, func(e sql.Expression) (sql.Expression, error) {
+		switch expr := e.(type) {
+		case *expression.UnresolvedFunction:
+			funcName := expr.Name()
+			builtInFunc, err := allBuiltInFuncs.Function(funcName)
+			if err != nil {
+				return nil, err
+			}
+			resolvedFunc, err := builtInFunc.Call(expr.Arguments...)
+			if err != nil {
+				return nil, err
+			}
+			return resolvedFunc, nil
+		case *expression.UnresolvedColumn:
+			//TODO: handle this
+			return nil, fmt.Errorf("columns in default are not yet supported")
+		default:
+			//TODO: explicitly handle all accepted expressions and reject in the default
+			return e, nil
+		}
+	})
+	if err != nil {
+		return nil, err
+	}
+	//TODO: currently (2+2)/2 will, when output as a string, give (2 + 2 / 2), which is clearly wrong
+	return sql.NewColumnDefaultValue(expr, isLiteral), nil
+}
+
+// MustStringToColumnDefaultValue is StringToColumnDefaultValue except that it panics on errors.
+func MustStringToColumnDefaultValue(ctx *sql.Context, exprStr string) *sql.ColumnDefaultValue {
```

Comment:
Is this test only? If so, put it in the test files (and maybe test packages)(

Reply:
It is not test only, any internal tables that contain default values that are created on engine creation make use of this.